ttnctl/util/payload_functions.go

```diff
@@ -0,0 +1,105 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/TheThingsNetwork/ttn/utils/parse"
+	"github.com/apex/log"
+	"github.com/robertkrimen/otto"
+)
+
+var (
+	ErrInvalidDecoderReturn       = errors.New("The decoder should return an object of fields")
+	ErrInvalidValidatorReturn     = errors.New("The validator should return a boolean")
+	ErrInvalidEncoderReturn       = errors.New("The encoder should return an array of a buffer of bytes")
+	ErrInvalidConverterReturn     = errors.New("The converter should return an object")
+	ErrInvalidPayloadFunctionType = errors.New("This type of payload function is not supported")
+	ErrUndefinedReturn            = errors.New("The function returned an undefined value")
+)
+
+func ValidatePayload(ctx log.Interface, code, payloadType string) (string, error) {
+	// We first parse the function to return if the AST cannnot be built (syntax error)
+	ctx.Info("Parsing function...")
+	err := parse.PayloadFunction(code)
+	if err != nil {
+		return "", err
+	}
+	ctx.Info("Function parsed successfully: syntax checked")
+
+	fmt.Printf("\n Test the function to detect runtime errors \n")
+	fmt.Println("Provide the signature of the payload function with test values")
+	fmt.Println(`
+Note:
+1) Use single quotes for strings: E.g: 'this is a valid string'
+2) Use the built-in function JSON.stringify() to provide json objects parameters: E.g: JSON.stingify({ valid: argument })
```

Comment:
Just give an example depending on the function type, this is not very user friendly